model/src/main/proto/exploration.proto

```diff
@@ -332,9 +350,30 @@ message HelpIndex {
     // have been revealed and there's no other help to provide. The boolean value here has no
     // importance and is always 'true'.
     bool everything_revealed = 3;
+
+    // Indicates this help index corresponds to the index of the next available hint within the hint
+    // list of a state.
+    int32 available_next_hint_index = 4;
+
+    // Indicates this help index corresponds to the index of the last revealed hint within the hint
+    // list of a state.
+    int32 latest_revealed_hint_index = 5;
   }
 }
 
+// A Structure corresponding to the state of hint and solution for the current exploration or
+// training session.
+message HintState {
+  // HelpIndex for the current state.
+  HelpIndex help_index = 1;
+
+  // Sequence number of the most recently scheduled task to show help to the learner.
+  int32 hint_sequence_number = 2;
+
+  // Delay for scheduling a new task to show more help to the learner.
+  int64 delay_to_show_next_hint_and_solution = 3;
+}
```

Comment:
I thought per our discussion only HelpIndex needs to be exposed--why is this needed?

Reply:
only help index is being exposed to the UI, these values are being stored in exploration Progress. we need to have these in the progress controller to correctly handler the timer that shows new help